### DISCUSSION TITLE
Chaste compile to shared library

### DISCUSSION THREAD

**Discussion by Elphikaus** [Chaste/Chaste#200] [Q&A] [answered]
I run chaste for cells, namely capsules.

I was wondering if chaste can be compiled easily into a shared library? I would like to try calling it from Julia.

**Comment by jmpf** [comment, marked as answer]
I think the answer's a firm "yes".

By default CMake makes each component/subfolder into a shared library and compiles shared libraries for all the component needed for the executables it's building:
```make TestChasteBuildInfo
...
[ 66%] Linking CXX shared library libchaste_global.so
[ 66%] Built target chaste_global
...
file ./global/libchaste_global.so
./global/libchaste_global.so: ELF 64-bit LSB shared object, x86-64, version 1 (GNU/Linux), dynamically linked, BuildID[sha1]=7dad2e1973a6433ac9aee3c5d956014384e22d59, with debug_info, not stripped
```
